### PR TITLE
[feature/MOB-560] Magic link login

### PR DESCRIPTION
**Pull request by franciscoaleixo**
**What does this PR do?**

   Adds support for magic link login.

**How should this be manually tested?**

  Check the same flows as vanilla for logging in with magic link.

**What are the relevant tickets?**

  [MOB-560](https://aptoide.atlassian.net/browse/MOB-560)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional tests pass

### CODE REVIEW

File: app/src/main/java/com/aptoide/uploader/UploaderApplication.java [hunk 1]
```diff
@@ -151,22 +159,29 @@ public AptoideAccountManager getAccountManager() {
...
+              getAptoideAuthenticationRx()),
           new SharedPreferencesAccountPersistence(PublishSubject.create(),
               PreferenceManager.getDefaultSharedPreferences(this), Schedulers.io()),
           new CredentialsValidator(), getSocialLogoutManager(), getAutoLoginPersistence());
     }
     return accountManager;
   }
 
+  public AptoideAuthenticationRx getAptoideAuthenticationRx() {
+    return new AptoideAuthenticationRx(new AptoideAuthentication(
```
Comment (phillipsgoncalves) [inline app/src/main/java/com/aptoide/uploader/UploaderApplication.java]: This is only used on one place but should we put it as singleton for prevention or not worth it ?

File: app/src/main/java/com/aptoide/uploader/account/AccountValidationException.java [hunk 2]
```diff
@@ -6,13 +6,14 @@
...
-  public static final int EMPTY_PASSWORD = 2;
-  public static final int EMPTY_STORE = 3;
+  public static final int EMPTY_CODE = 2;
+  public static final int EMPTY_EMAIL_AND_CODE = 3;
   public static final int INVALID_PASSWORD = 4;
   public static final int EMPTY_EMAIL_AND_PASSWORD = 5;
   public static final int EMPTY_EMAIL_AND_STORE = 6;
   public static final int EMPTY_PASSWORD_AND_STORE = 7;
   public static final int EMPTY_EMAIL_PASSWORD_AND_STORE = 8;
+  // TODO: 6/4/20 I replaced one that was alone with the empty store add it in case needed.
```
Comment (phillipsgoncalves) [inline app/src/main/java/com/aptoide/uploader/account/AccountValidationException.java]: todo note

File: app/src/main/java/com/aptoide/uploader/account/AptoideAccountManager.java [hunk 3]
```diff
@@ -21,9 +23,15 @@ public AptoideAccountManager(AccountService accountService, AccountPersistence a
...
 
-  public Completable login(String username, String password) {
-    return accountService.getAccount(username, password)
+  public Completable login(AptoideCredentials aptoideCredentials) {
+    return credentialsValidator.validate(aptoideCredentials)
+        .andThen(
+            accountService.getAccount(aptoideCredentials.getEmail(), aptoideCredentials.getCode(),
+                aptoideCredentials.getState(), aptoideCredentials.getAgent()))
         .flatMapCompletable(account -> accountPersistence.save(account));
+    // TODO: 6/4/20 one error we had on vanilla was the fact that
```
Comment (phillipsgoncalves) [inline app/src/main/java/com/aptoide/uploader/account/AptoideAccountManager.java]: todo comment again

File: app/src/main/java/com/aptoide/uploader/account/AptoideAccountManager.java [hunk 4]
```diff
@@ -36,13 +44,23 @@ public Completable loginWithFacebook(String email, String serverAuthToken) {
...
+  }
+
   public Observable<Account> getAccount() {
     return accountPersistence.getAccount();
   }
 
   public Completable create(String email, String password, String storeName) {
-    return credentialsValidator.validate(email, password, storeName)
-        .andThen(accountService.createAccount(email, password, storeName))
+    // TODO: 6/4/20 commented this validation in order to not have to change the call of this method as it is going to probably be removed?
```
Comment (phillipsgoncalves) [inline app/src/main/java/com/aptoide/uploader/account/AptoideAccountManager.java]: Todo comment again. 
Also, this is related to a validation that was being done before and is not being done now. Was this validated with John ?

File: app/src/main/java/com/aptoide/uploader/account/CredentialsValidator.java [hunk 5]
```diff
@@ -1,13 +1,18 @@
...
+import io.reactivex.Single;
 
 /**
  * Created by jose_messejana on 05-01-2018.
  */
 
 public class CredentialsValidator {
 
+  /*
   public Completable validate(String email, String password, String storeName) {
```
Comment (phillipsgoncalves) [inline app/src/main/java/com/aptoide/uploader/account/CredentialsValidator.java]: Related with the previous comment. This validation was commented, does it need to be done? Was this validated with joao

File: app/src/main/java/com/aptoide/uploader/account/network/RetrofitAccountService.java [hunk 6]
```diff
@@ -36,29 +38,18 @@
...
-            new AccountRequestBody(Collections.singletonList("meta"), accessToken))
-            .singleOrError())
-        .flatMap(response -> {
-          if (response.isSuccessful() && response.body()
-              .isOk()) {
-            return Single.just(mapper.map(response.body(), BaseAccount.LoginType.APTOIDE));
-          }
-          return Single.error(new IllegalStateException(response.message()));
-        });
+    this.aptoideAuthentication = aptoideAuthentication;
```
Comment (phillipsgoncalves) [inline app/src/main/java/com/aptoide/uploader/account/network/RetrofitAccountService.java:41]: There is code in this class still related with the old login, like the createAccountMethod that has null password and storename being passed to it, which is being used in createAccount presenter, which is not being really used anymore i think.
Remove the rest of unused code

File: app/src/main/java/com/aptoide/uploader/account/view/AccountErrorMapper.java [hunk 7]
```diff
@@ -34,12 +34,12 @@ public String map(Throwable throwable) {
       case AccountValidationException.EMPTY_EMAIL:
         message = context.getString(R.string.no_email_error_message);
         break;
-      case AccountValidationException.EMPTY_PASSWORD:
+      /*case AccountValidationException.EMPTY_PASSWORD:
```
Comment (phillipsgoncalves) [inline app/src/main/java/com/aptoide/uploader/account/view/AccountErrorMapper.java]: commented code related with the previously mentioned verification

File: app/src/main/java/com/aptoide/uploader/account/view/LoginPresenter.java [hunk 8]
```diff
@@ -118,13 +118,6 @@ public LoginPresenter(LoginView view, AptoideAccountManager accountManager,
           view.showNetworkError();
           uploaderAnalytics.sendLoginEvent("facebook", "fail");
         }));
-
-    compositeDisposable.add(view.getLifecycleEvent()
-        .filter(lifecycleEvent -> lifecycleEvent.equals(View.LifecycleEvent.CREATE))
-        .flatMap(__ -> view.clickOnBlog())
-        .doOnNext(__ -> loginNavigator.openBlogUrl())
```
Comment (phillipsgoncalves) [inline app/src/main/java/com/aptoide/uploader/account/view/LoginPresenter.java:125]: This was removed, so this method on login navigator is not being used anymore

**Review by phillipsgoncalves — CHANGES_REQUESTED** [review]
Also regarding the strings, is everything ok with carlos ?

Also, clicking on the secure message is not navigating to the blog. Was it supposed to?

File: app/src/main/java/com/aptoide/uploader/account/AptoideAccountManager.java (hunk 4 above)
Comment (franciscoaleixo) [reply app/src/main/java/com/aptoide/uploader/account/AptoideAccountManager.java]: Removed this and their usage.

**Pull request merged**